src/main/resources/xlr_xldeploy/XLDVersionsTile.py

```diff
@@ -11,14 +11,29 @@
 
 from xlr_xldeploy.XLDeployClientUtil import XLDeployClientUtil
 
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+
 if not xldeployServer:
     raise Exception("XL Deploy server ID must be provided")
 
+#check the range of values for numberOfDays,
+if not 1 <= numberOfDays <= maxDays:
+    numberOfDays = maxDays
+
 xld_client = XLDeployClientUtil.create_xldeploy_client(xldeployServer, username, password)
 if xld_client.check_ci_exist(environment):
     if date:
-        data = xld_client.get_deployed_applications_for_environment(environment, date)
+        #if date provided, set a begin_date based on provided date - number of days
```

Comment:
There's now quite some duplication between the two branches... Consider refactoring it.